woocommerce/woocommerce-ios
# Order Details > variation details: storage layer changes

Storage layer changes for #2280

This PR only contains changes in the storage layer, since the Networking/Yosemite changes are going to be huge as well 🙂 

## Changes

- Schema updates (please review the diffs on model 35 in https://github.com/woocommerce/woocommerce-ios/commit/2c63d3fcef2155e2fdda64909343faa6506c275c#diff-2f9c7ea9bf9793125e3c162b4fcb6dbd5ff9c50054f30aacfc2b996b9c11de71)
  - Added `OrderItemAttribute` entity
  - Added  `attributes: [OrderItemAttribute]` relationship to `OrderItem`
- Added a unit test for migrating from mode 34 to 35 in `MigrationTests`

## Testing

Please run a confidence check on migration!

- Launch the app from `develop` and log in or stay logged in
- Stop the app
- Launch the app from PR branch --> the app should launch in logged in state with the same data

Update release notes:
